### PR TITLE
fixes #22062 - support vmware vmrc console

### PR DESCRIPTION
**Pull request by timogoebel**
As vSphere >= 6.5 dropped support for the VNC consoles, this is a second attempt (#4755) to provide an alternative.
This uses the VMRC type console, which is basically a standalone program (or something like VMWare Fusion) that provides the console, i.e. the console is not displayed in the browser.
This is actually quite nice (feature wise) because you can easily attach .iso images etc.

image

image
"Open Link in VMWare Fusion?"

image

### CODE REVIEW

File: app/views/hosts/console/vmrc.html.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<% title "#{@console[:name]}" %>
+<div class="blank-slate-pf">
+  <div class="blank-slate-pf-icon">
+    <%= icon_text('screen', '', :kind => 'pficon') %>
+  </div>
+  <h1><%= _('VMRC Console') %></h1>
+  <p><%= _('To launch the VMRC console, you need the VMRC software installed.') %></p>
```
Comment (ohadlevy) [inline app/views/hosts/console/vmrc.html.erb:7]: I assume that there is no way for us to detect it? (if its installed or not?)

**Review by ohadlevy — APPROVED** [review]
Tested and looks good, overall working nicely, I only had one question.

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 2]
```diff
@@ -537,6 +564,18 @@ def associated_host(vm)
       associate_by("mac", vm.interfaces.map(&:mac))
     end
 
+    def display_type
+      attrs[:display] || 'vmrc'
```
Comment (ohadlevy) [inline app/models/compute_resources/foreman/model/vmware.rb:568]: does it make sense to always default to vmrc? (all CR will default to that on first time?) do we have vmware version number so we can make an educated decision?

File: app/views/hosts/console/vmrc.html.erb (hunk 1 above)
Comment (timogoebel) [reply app/views/hosts/console/vmrc.html.erb:7]: Yes, as far as I know, we can't detect if it's installed or not.

File: app/models/compute_resources/foreman/model/vmware.rb (hunk 2 above)
Comment (timogoebel) [reply app/models/compute_resources/foreman/model/vmware.rb:568]: Maybe we should add a migration to set all existing compute resources to `vnc` as that was the default behavior. As `vmrc` works with all compute resources, I think it's the preferred choice. We can get the vmware version number, but that requires an established connection and I think it's not worth the effort.

**Comment by ohadlevy** [comment]
Thanks!

**Pull request merged**